File: MituWork/WeixinFramework/Study/VbNetSnapIn/VbNetSnapIn/Class1.vb
Imports System.Windows.Forms
Imports CommonSnappable

<CompanyInfo(CompanyName:="Chuky's Software", CompanyUrl:="www.ChuckySoft.com")>
Public Class VbSnapIn
    Implements IAppFunctionality

    Public Sub DoIt() Implements IAppFunctionality.DoIt
        MessageBox.Show("You have just used the VB snap in!")
    End Sub
End Class

File: MituWork/WeixinFramework/Study/CSharpSnapIn/CSharpSnapIn/Class1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using CommonSnappable;
using System.Windows.Forms;

namespace CSharpSnapIn
{
    [CompanyInfo(CompanyName = "My Company", CompanyUrl = "www.MyCompany.com")]
    public class CSharpModule : IAppFunctionality
    {
        //显式实现接口，把DoIt()方法隐藏起来
        void IAppFunctionality.DoIt()
        {
            MessageBox.Show("You have just used the C# snap un!");
        }
    }
}
